sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/PartitionReceiverLiveTests.cs

```diff
@@ -36,31 +41,1340 @@ public class PartitionReceiverLiveTests
+                    }
+                }
+
+                cancellationSource.Cancel();
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="PartitionReceiver" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ReceiverCanReadFromEnqueuedTime()
+        {
+            await using (EventHubScope scope = await EventHubScope.CreateAsync(1))
+            {
+                using var cancellationSource = new CancellationTokenSource();
+                cancellationSource.CancelAfter(TimeSpan.FromMinutes(4));
+
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+                var partition = (await QueryPartitionsAsync(connectionString, cancellationSource.Token)).First();
+                var seedEvents = EventGenerator.CreateEvents(50).ToList();
+                var sourceEvents = EventGenerator.CreateEvents(100).ToList();
+
+                // Seed the partition with a set of events prior to reading.  When the send call returns, all events were
+                // accepted by the Event Hubs service and should be available in the partition.  Provide a minor delay to
+                // allow for any latency within the service.
+                //
+                // Once sent, query the partition and determine the offset of the last enqueued event, then send the new set
+                // of events that should appear after the starting position.
+
+                DateTimeOffset lastEnqueuedTime;
+                EventPosition startingPosition;
+
+                await using (var producer = new EventHubProducerClient(connectionString))
+                {
+                    await SendEventsAsync(connectionString, seedEvents, new CreateBatchOptions { PartitionId = partition }, cancellationSource.Token);
+                    await Task.Delay(250);
+
+                    lastEnqueuedTime = (await producer.GetPartitionPropertiesAsync(partition, cancellationSource.Token)).LastEnqueuedTime;
+                    startingPosition = EventPosition.FromEnqueuedTime(lastEnqueuedTime);
+
+                    await SendEventsAsync(connectionString, sourceEvents, new CreateBatchOptions { PartitionId = partition }, cancellationSource.Token);
+                }
+
+                // Read the events and validate the resulting state.
+
+                await using (var receiver = new PartitionReceiver(EventHubConsumerClient.DefaultConsumerGroupName, partition, startingPosition, connectionString))
+                {
+                    var readState = await ReadEventsAsync(receiver, sourceEvents.Count, cancellationSource.Token);
+                    Assert.That(cancellationSource.IsCancellationRequested, Is.False, "The cancellation token should not have been signaled.");
```

Comment:
Should there be an assert here validating we received events according to the expected `startingPosition`?

Reply:
Lines 614 - 619 walk the `sourceEvents` which were those sent after the starting position was read from the last of the `seedEvents` that were sent.   We're validating that we receive every event after the enqueue time.    

That said, we were missing a comparison of counts to ensure that we didn't get extraneous events;  I've added that in.

Reply:
Ported that check over to the consumer test scenario as well.